api/server/server.go

```diff
@@ -196,13 +196,14 @@ func postContainersKill(eng *engine.Engine, version version.Version, w http.Resp
 	if err := parseForm(r); err != nil {
 		return err
 	}
-	job := eng.Job("kill", vars["name"])
-	if sig := r.Form.Get("signal"); sig != "" {
-		job.Args = append(job.Args, sig)
-	}
-	if err := job.Run(); err != nil {
+
+	name := vars["name"]
+	sig := vars["signal"]
```

Comment:
I think the signal should be converted to an int in the API layer so that a StatusBadRequest can be returned for things like this as `int` is the final type for this var.